condition_resource_contains_test.go

```diff
@@ -0,0 +1,71 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ladon
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResourceContains(t *testing.T) {
+	for _, c := range []struct {
+		matches   string
+		delimiter string
+		resource  string
+		value     string
+		pass      bool
+	}{
+		//Correct matching within resource string without delimiter:
+		{matches: "Exact resource component match in resource string, with no delimiter", value: ":GroupId:2:", resource: "resources:sensor-data:GroupId:2:deviceIds:a9b576e8-7419-4eed-a010-7f68ec0ff588", pass: true},
```

Comment:
There are test cases missing here. I would like to have the following covered:

* With delimiter (`:`):
  * Tests if it passes when the value is a prefix (value: `foo:bar`, resource: `foo:bar:baz`)
  * Tests if it fails when the value is not a prefix  (value: `foo:bar`, resource: `foo:baz:bar`)
  * Tests if it passes when the value is an affix (value: `foo:bar`, resource: `baz:foo:bar`)
  * Tests if it fails when the value is not an affix (value: `foo:baz`, resource: `baz:foo:baz`)
  * Tests if it passes when the value is somewhere in the middle (value: `foo`, resource: `baz:foo:bar`)
  * Tests if it fails when the value is not somewhere in the middle (value: `foo`, resource: `baz:foo:baz`)
* Without delimiter -> same test cases as above but without delimiter, obviously :)

We should also decide what happens when `{delimiter: ":", value: ":foo:", resource: "bar:foo:bar"`. Right now this would fail. Should it?

Everything else can be thrown out - unless I missed something.

Reply:
much obliged, will update.

Reply:
Tests if it fails when the value is not somewhere in the middle (value: foo, resource: baz:foo:baz) is meant to be foo:bar I am guessing

Reply:
Yes, the resource should be something like `baz:faa:baz` :)